api/index.php

```diff
@@ -40,7 +40,10 @@ function api()
 
     $router = ApiRouter::get_router();
 
-    api_output_response($router->route($path, $query_params));
```

Comment:
Can you elaborate on this change and the related changes to all the functions in `v1_projects.inc` to take a 3rd parameter?

Reply:
Away from home at the momennt but it was to enable testing. There could be a better way to do it with stub or something like that in phpunit.

Reply:
I've had a look at this:

Setting a value for method: this comes from $_SERVER["REQUEST_METHOD"] so we could just set this global variable. This works but is not a good thing to do. If I understand correctly, namespaces will not help because super globals are visible in all contexts, so I think adding a parameter to route() could be the best option for this since it is a very simple change. The code already passes the value for $query_params through route().

Setting a value for api_get_request_body(). This function in index.php is not visible in the tests so we can just make a new global function for it and manipulate the value it returns by a global variable.